navigator/runtime-experimental/src/main/java/com/freeletics/mad/navigator/compose/internal/Stack.kt

```diff
@@ -0,0 +1,139 @@
+package com.freeletics.mad.navigator.compose.internal
+
+import android.os.Bundle
+import androidx.core.os.bundleOf
+import com.freeletics.mad.navigator.BaseRoute
+import com.freeletics.mad.navigator.NavRoot
+import com.freeletics.mad.navigator.NavRoute
+import com.freeletics.mad.navigator.compose.ContentDestination
+import com.freeletics.mad.navigator.compose.ScreenDestination
+import com.freeletics.mad.navigator.internal.DestinationId
+import com.freeletics.mad.navigator.internal.destinationId
+import java.util.UUID
+
+internal class Stack private constructor(
+    initialStack: List<StackEntry<*>>,
+    private val destinations: List<ContentDestination<*>>,
+    private val onStackEntryRemoved: (StackEntry.Id) -> Unit,
+    private val idGenerator: () -> String,
+) {
+    private val stack = ArrayDeque<StackEntry<*>>(20).also {
```

Comment:
Is there a reason for size `20`?

Reply:
I just chose a number that should be big enough for most back stacks but also not too big.